src/SystemManager.cc

```diff
@@ -101,6 +106,57 @@ size_t SystemManager::ActivatePendingSystems()
   return count;
 }
 
+//////////////////////////////////////////////////
+void SystemManager::Reset(const UpdateInfo &_info, EntityComponentManager &_ecm)
+{
+  {
+    std::lock_guard<std::mutex> lock(this->pendingSystemsMutex);
+    this->pendingSystems.clear();
+  }
+
+  // Clear all iterable collections of systems
+  this->systemsConfigure.clear();
+  this->systemsReset.clear();
+  this->systemsPreupdate.clear();
+  this->systemsUpdate.clear();
+  this->systemsPostupdate.clear();
+
+  this->entityCompMgr = &_ecm;
```

Comment:
The ECM used to construct `SystemManager` should be the same as the `_ecm` here. Do we need to assign it here? Looks harmless, but I thought it might cause confusion later on.